src/compile/model.ts

```diff
@@ -169,12 +183,12 @@ export abstract class Model {
     // FIXME: write assembleScales() in scale.ts that
     // help assemble scale domains with scale signature as well
     return flatten(vals(this.component.scale).map((scales: ScaleComponents) => {
-      let arr = [scales.main];
+      let arr = [renameScaleData(this, scales.main)];
```

Comment:
Maybe add comment that we need to rename because we already put these data name in the parse phase and the data source name might be renamed after that.   (Do I understand correctly?)